### PR TITLE
remove 'component' from generated modules and tests

### PR DESCRIPTION
**Pull request by bocekm**
And also from UI and man page.

This PR continues upon the already merged PR https://github.com/upgrades-migrations/preupgrade-assistant/pull/195.
Resolves: [rhbz#1372871](https://bugzilla.redhat.com/show_bug.cgi?id=1372871)

### CODE REVIEW

**Comment by bocekm** [comment]
@AloisMahdal Another issue emerged regarding this 'easy to fix' RFE. The UI is supposed to show a history of generated assessment reports. As @pirat89 found out, it seems that with this change the UI-related processing will not be able to parse the old reports correctly (old in terms of having 'component').

If you're not able to test the UI to work correctly with both old and new reports this week, I'll drop this PR and also revert the previously merged PR #195. Then it can wait in non-master branch to be resurrected once we have more time to play with RFEs.

**Comment by AloisMahdal** [comment]
Nice catch.  I'll be happy to test UI as well.

**Comment by phracek** [comment]
By the way, What is the reason to remove 'component' from modules. Till now, each message is identified by component name and if it is wrong, like 'unknown' then module has to be fixed. The message should be somehow identified and we have to find out a way, how to identify it.

I would prefer to focus to important things like #193, #184, #165, #131, #146.

**Comment by pirat89** [comment]
@phracek  Just my mini input, because I don't remember it completely.
I guess it is really mainly because of unkown component. It's wrong identificator because many modules can't be assigned to any specific component. If there has to be identificator, it should be module ID itself or just print info message that a module is now processed before run of the module, so it will be clear which module prints these logs.

**Comment by bocekm** [comment]
@phracek This was supposed to be a quick and straightforward change, having little or no effect on the existing functionality. I agree we should focus on more important issues. The best would be to revert the previous pull request PR #195.
If we are about to keep some message identification in the log, it should be the module name, as @pirat89 suggests.

**Comment by bocekm** [comment]
FYI, this affects only INFO|ERROR|WARNING messages, not RISK messages. When I look into */var/log/preupgrade/preupg.log*, the INFO|ERROR|WARNING messages are not there anyway. They are just in the result.xml/html.

**Comment by phracek** [comment]
@pirat89 If there are some unknown component names, then modules should be update properly. This is only information you do not specify 'COMPONENT' in your module.

**Comment by AloisMahdal** [comment]
@phracek  The reasoning behind this is in rhbz#1372871.  In short, the component part is not necessary (it's implied by module and/or message text), makes log parsing harder and is never used by any module anyway.

Also, while functional impact is not so significant, one of reasons for this is also to aid the *cleanup* goal by getting rid of unnecessary code: the idea is that by every (smaller, more focused and testable) step like this, we can shave off a little bit of complexity that makes refactoring so risky and painful.

Regarding processing old vs. new reports in UI: We could support both formats.  However I highly doubt that we have been keeping compatibility between reports, nobody ever tested this AFAIK.  Also, given that the rule is to always use latest, freshest possible assessments (with official upgrade path this is even a hard requirement), the scenario where user needs to view an old report next to new one in UI does not seem very likely to me.

File: man/preupgrade-assistant-api.1 [hunk 1]
```diff
@@ -110,11 +110,11 @@ The levels None, Slight and Medium change return code exit_fail to exit_needs_in
 
 There are several functions which do logging:
 
-\fBlog_{debug,info,warning,error} <component> <message>\fP
+\fBlog_{debug,info,warning,error} <message>\fP
 
 The function creates logs in the format:
 
-<SEVERITIES> <component> <TIMESTAMP> <MESSAGE>
+<SEVERITIES> <TIMESTAMP> <MESSAGE>
```
Comment (bocekm) [inline man/preupgrade-assistant-api.1]: This is actually not true, there's no timestamp anymore. The result is that the log messages are not shown in UI in the current state and will not be after this update.

**Comment by pirat89** [comment]
@phracek Which component should I set for module that serves 1000 rpms? Or set of all libraries? Or when module solves troubles for just two component... etc. There is not any specific component for such modules. And we have many modules like that. Should I sest "distribution"? In that case the component doesn't have any usable information and "distribution" has same benefit as "unkown".

Agree with Lojza in this.

**Comment by bocekm** [comment]
We tested this change with @AloisMahdal and it has no impact on UI.

**Pull request merged**